r/src/buffer.cpp

```diff
@@ -0,0 +1,44 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "arrow_types.h"
+
+// [[Rcpp::export]]
+bool Buffer__is_mutable(const std::shared_ptr<arrow::Buffer>& buffer) {
+  return buffer->is_mutable();
+}
+
+// [[Rcpp::export]]
+void Buffer__ZeroPadding(const std::shared_ptr<arrow::Buffer>& buffer) {
+  buffer->ZeroPadding();
+}
+
+// [[Rcpp::export]]
+int64_t Buffer__capacity(const std::shared_ptr<arrow::Buffer>& buffer) {
+  return buffer->capacity();
+}
+
+// [[Rcpp::export]]
+int64_t Buffer__size(const std::shared_ptr<arrow::Buffer>& buffer) {
+  return buffer->size();
+}
+
+// [[Rcpp::export]]
+std::shared_ptr<arrow::Buffer> Buffer__parent(
+    const std::shared_ptr<arrow::Buffer>& buffer) {
+  return buffer->parent();
+}
```

Comment:
Maybe better to not wrap this method unless you need it. I've been considering changing the way that slice references work. Can always remove this later, too

Reply:
Was mostly a because it's there, we don't really need it.